### ISSUE TITLE
`filled()` / `blank()` stub over-narrows: string input collapses to literal false

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#751]
## Description

The `filled()` and `blank()` helper stubs in `stubs/common/Support/helpers.stubphp` use conditional return types that over-narrow when the input type is `string` (or `?string`), collapsing the return type to literal `false`. This triggers `DocblockTypeContradiction` ("Operand of type false is always falsy") on idiomatic `if (filled($someMethodReturningNullableString())) { ... }` checks.

## Reproduction

```php
trait CanCallParentAction
{
    public function getParentActionCallLivewireClickHandler(): ?string
    {
        return $this->parentActionCallLivewireClickHandler;
    }
}

// Elsewhere:
if (filled($handler = $this->getParentActionCallLivewireClickHandler())) {
    // DocblockTypeContradiction: Operand of type false is always falsy
    $handler .= '(';
    // ...
}
```

## Root cause

[`stubs/common/Support/helpers.stubphp:77-85`](https://github.com/psalm/psalm-plugin-laravel/blob/master/stubs/common/Support/helpers.stubphp#L77-L85):

```php
/**
 * @psalm-return (\$value is (bool|numeric|non-empty-array)
 *  ? true
 *  : (\$value is (null|''|array<never, never>) ? false : bool)
 * )
 */
function filled(\$value) {}
```

When `\$value` is inferred as `string|null`:
- `null` branch: matches `null|''|...` → returns `false`
- `string` branch: Psalm sees `''` as a subtype of `string`, so the `string is (null|''|...)` check evaluates in the affirmative → returns `false`
- Combined result: literal `false`

The stub for `blank()` (lines 31-40) has the same shape with the same issue.

## Impact

In filamentphp/filament (audited with plugin v4.8.2, `errorLevel=1`), **245 of 476** `DocblockTypeContradiction` errors (52%) come from this root cause. It also affects any codebase that uses `filled(\$nullableString)` / `blank(\$nullableString)` as control-flow guards, which is idiomatic Laravel.

## Expected behavior

`filled(string|null)` should return `bool`, not `false`. The narrowed `true`/`false` returns should only apply when Psalm can prove the input is exactly in the narrow set (e.g., `non-empty-string` → `true`, `''` literal → `false`).

## Proposed fix

Replace the conditional return type with assertion annotations (the approach Larastan uses and which Psalm handles correctly):

```php
/**
 * @param  mixed  \$value
 * @psalm-assert-if-true !null \$value
 * @psalm-assert-if-false null|''|array<never, never> \$value
 */
function filled(\$value): bool {}
```

Or if conditional return types are preferred, tighten the narrow clause to require `non-empty-string` instead of any `string`:

```
(\$value is non-empty-string|non-empty-array|numeric|true
 ? true
 : (\$value is null|''|false|array<never, never>
    ? false
    : bool))
```

The same fix applies to `blank()` (inverted).

## Related

- Stub location: `stubs/common/Support/helpers.stubphp:31-40` (blank), `:77-85` (filled)
- Surfaces in any Laravel codebase using `filled(\$nullableString)` as a guard (common in Filament, Nova, Horizon, etc.)
